include/storage/flash_map.h

```diff
@@ -58,6 +59,38 @@ struct flash_area {
 	size_t fa_size;
 };
 
+/*
+ * The getter functions cover fields of flash_area from mcuboot not the internal implementation
+ * of Zephyr.
+ */
+static inline off_t flash_area_get_off(const struct flash_area *fa)
```

Comment:
Do we really need any of these inside the zephyr repository itself?

All the other MCUboot ports except mynewt seem to put their flash area abstractions inside MCUboot:

https://github.com/mcu-tools/mcuboot/blob/main/boot/cypress/cy_flash_pal/cy_flash_map.c
https://github.com/mcu-tools/mcuboot/blob/main/boot/mbed/src/flash_map_backend.cpp

Mynewt is of course the exception because these APIs were already defined using the mynewt API.

Following the other ports by putting the "glue" between the zephyr native APIs and MCUboot's "mynewt style" view of the world seems better than putting the glue here in the zephyr API header.

Zephyr apps don't seem to need these definitions; they can access the structure directly if they are using the zephyr flash map API.

Am I missing something?

Reply:
No, you are not missing anything. I will move the functions.